internal/oci/mutate/mutate.go

```diff
@@ -121,3 +122,81 @@ func (i *indexWrapper) SignedImageIndex(h v1.Hash) (oci.SignedImageIndex, error)
 		return signed.ImageIndex(unsigned), nil
 	}
 }
+
+// SignEntity attaches the provided signature to the provided entity.
+func SignEntity(se oci.SignedEntity, sig oci.Signature, opts ...SignOption) (oci.SignedEntity, error) {
```

Comment:
Let's replace things like `SignFoo` with `AttachSignatureToFoo` (or something) to avoid confusing these with the objects doing cryptographic operations. Think that "sign" conceptual space is already crowded enough in this library ecosystem

Reply:
Sure, I like the `Attach` nuance.  Let me just battle merge conflicts for a while 😉